lib/elixir/lib/enum.ex

```diff
@@ -4462,6 +4462,7 @@ defmodule Enum do
 
       {:ok, count, fun} when is_function(fun, 1) ->
         amount = Kernel.min(amount, count - start)
+        amount = if step == 1, do: amount, else: div(amount - 1, step) + 1
```

Comment:
Note: this is the same logic used in other branches above

Reply:
(maybe a function would be nice even if it is just one line, WDYT?)

Reply:
Up to you!